rpc/src/rpc.rs

```diff
@@ -6099,12 +6131,27 @@ pub mod tests {
         );
         let res = io.handle_request_sync(&req, meta.clone());
         let expected = json!({
-            "jsonrpc":"2.0",
-            "error": {
-                "code": -32003,
-                "message": "Transaction signature verification failure"
+            "jsonrpc": "2.0",
+            "result": {
+                "context": { "slot": 0, "apiVersion": RpcApiVersion::default() },
+                "value": {
+                    "accounts": null,
+                    "err": "SignatureFailure",
+                    "innerInstructions": null,
+                    "loadedAccountsDataSize": 0,
+                    "fee": null,
+                    "loadedAddresses": { "readonly": [], "writable": [] },
```

Comment:
Note, for posterity: It's pretty annoying that in `simulate_transaction()` this gets set to `{ "readonly": [], "writable": [] }` upon a signature failure, but in `send_transaction()` it gets set to `null`.